lib/cc/engine/analyzers/ruby/main.rb

```diff
@@ -24,6 +24,10 @@ class Main < CC::Engine::Analyzers::Base
 
           def process_file(file)
             RubyParser.new.process(File.binread(file), file, TIMEOUT)
+          rescue Timeout::Error
+            warn "TIMEOUT parsing #{file}. Skipping."
+          rescue => e
```

Comment:
1. Those rescues were actually necessary to get the code running on MRI, so I'd like to leave them.
2. Given permission from @GordonDiggs... I'll do the followup PR to clean this up and get error handling generally better. Otherwise I'll open an issue pointing here so we have an artifact to track.